Parser: bare return statement causes function block to remain unclosed

## Description

A bare `return` statement (without a return value) causes the parser to incorrectly leave the function block unclosed. This results in subsequent function declarations being parsed as if they were inside the previous function.

## Reproduction

```ez
do main(){
    println("test")
}

do foo() {
    return
}

do bar() {
    println("after foo")
}
```

**Expected:** All three functions parse correctly.

**Actual:**
```
error[E2019]: function declarations are not allowed inside functions
  --> test.ez:9:1
   |
9 | do bar() {
   | ^^ function inside function not allowed
   |

error[E2004]: unclosed brace - expected '}'
  --> test.ez:5:10
   |
5 | do foo() {
   |          ^ missing closing brace
```

## Root Cause

In `parseReturnStatement()` (pkg/parser/parser.go:903):

```go
func (p *Parser) parseReturnStatement() *ReturnStatement {
    stmt := &ReturnStatement{Token: p.currentToken}
    stmt.Values = []Expression{}

    p.nextToken()  // Advances past 'return'

    // Check for empty return
    if p.currentTokenMatches(RBRACE) || p.currentTokenMatches(EOF) {
        return stmt  // Returns with currentToken == RBRACE
    }
    // ...
}
```

For a bare `return` followed by `}`:
1. `parseReturnStatement` advances to the `}` and returns with `currentToken = }`
2. The block parsing loop at line 992 does `p.nextToken()`, advancing **past** the `}`
3. The loop condition `!p.currentTokenMatches(RBRACE)` is now true (we're past the brace)
4. The loop continues, treating subsequent code as being inside the function

## Suggested Fix

Check `peekToken` **before** advancing, so empty returns don't consume the closing brace:

```go
func (p *Parser) parseReturnStatement() *ReturnStatement {
    stmt := &ReturnStatement{Token: p.currentToken}
    stmt.Values = []Expression{}

    // Check for empty return BEFORE advancing
    if p.peekTokenMatches(RBRACE) || p.peekTokenMatches(EOF) {
        return stmt  // Stay at 'return' token
    }

    p.nextToken() // Only advance if there's a value to parse

    // Parse return values
    stmt.Values = append(stmt.Values, p.parseExpression(LOWEST))
    // ...
}
```

## Impact

Any function with a bare `return` statement will cause parsing failures for all subsequent top-level declarations in the file.

--- a/pkg/parser/parser.go
+++ b/pkg/parser/parser.go
@@ -904,12 +904,13 @@ func (p *Parser) parseReturnStatement() *ReturnStatement {
 	stmt := &ReturnStatement{Token: p.currentToken}
 	stmt.Values = []Expression{}
 
-	p.nextToken()
-
-	// Check for empty return
-	if p.currentTokenMatches(RBRACE) || p.currentTokenMatches(EOF) {
+	// Check for empty return BEFORE advancing
+	// This prevents consuming the closing brace which would leave the block unclosed
+	if p.peekTokenMatches(RBRACE) || p.peekTokenMatches(EOF) || p.peekTokenMatches(NEWLINE) {
 		return stmt
 	}
+
+	p.nextToken() // Only advance if there's a value to parse
 
 	// Parse return values
 	stmt.Values = append(stmt.Values, p.parseExpression(LOWEST))

